aws/external/shared_config.go

```diff
@@ -105,6 +108,21 @@ type SharedConfig struct {
 	//
 	// s3_use_arn_region=true
 	S3UseARNRegion *bool
+
+	// EnableEndpointDiscovery can be enabled in the shared config by setting
+	// endpoint_discovery_enabled to true
+	//
+	//	endpoint_discovery_enabled = true
+	EnableEndpointDiscovery *bool
+}
+
+// GetEnableEndpointDiscovery returns whether to enable service endpoint discovery
+func (c *SharedConfig) GetEnableEndpointDiscovery() (value, ok bool, err error) {
```

Comment:
Will be helper to ensure the `SharedConfig` and `EnvConfig` can satisfy the `EnableEndpointDiscoveryProvider` interface via test type assertion.